src/ethereum/trace.py

```diff
@@ -167,40 +167,29 @@ class GasAndRefund:
 """
 
 
-def discard_evm_trace(
-    evm: object,
-    event: TraceEvent,
-    trace_memory: bool = False,
-    trace_stack: bool = True,
-    trace_return_data: bool = False,
-) -> None:
+@dataclass
+class BaseTrace:
     """
-    An [`EvmTracer`] that discards all events.
-
-    [`EvmTracer`]: ref:ethereum.trace.EvmTracer
+    A basic trace.
     """
 
 
-class EvmTracer(Protocol):
+@dataclass
+class BaseEvmTracer:
     """
-    [`Protocol`] that describes tracer functions.
-
-    See [`ethereum.trace`] for details about tracing in general, and
-    [`__call__`] for more on how to implement a tracer.
-
-    [`Protocol`]: https://docs.python.org/3/library/typing.html#typing.Protocol
-    [`ethereum.trace`]: ref:ethereum.trace
-    [`__call__`]: ref:ethereum.trace.EvmTracer.__call__
+    Default tracer for an EVM.
     """
 
-    def __call__(
+    traces: list[BaseTrace] = field(default_factory=list)
+    trace_memory: bool = False
+    trace_stack: bool = True
+    trace_return_data: bool = False
+    output_basedir: str = "."
```

Comment:
This class doesn't use these fields, so why does it define them? For `traces` in particular, we should consider a future implementation that fixes #909 (and so `traces` in the base class makes no sense).